### DISCUSSION TITLE
Does not appear to be a way to get the actual screen mute state?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [WesSouza/lgtv-ip-control#124] [General]
Does not appear to be a way to get the actual screen mute state?  No easy API call like say getMuteState() or such.  I am starting to look at the WebOS documentation to see if I can find any reference to same.

Right now I have my NodeRed flow 'learn' the screen state, by just remembering the on or off state.  .  I am only using 'allmuteoff' and ''screenmuteoff' as a toggle.

Of course this is requires the operator to select the screen toggle twice if the screen is off, but the NodeRed flow thinks the state is on, but since iit is a simple toggle, works with note expectation.  Only hove to go through the effort if NodeRed resets or reloads, and the actual screen mute state has changed since last use, say via remote or CLI for example.

I am posting this discussion in hopes some WebOS developer comes across this discussion and provides some insight.

**Comment by jadoc** [comment]
I believe the screen mute feature isn't intended for consumers, but for public displays where the TV is being controlled entirely by remote automation. They assume the remote control software is intelligent enough to know/remember what state it indented for the TV to be in.

Generally, querying the current TV state is only necessary when driving interactive user interfaces.

**Comment by Jibun-no-Kage** [comment]
Yes, I pretty much came to that realization, and established a control state, where the code 'learns' of the screen state, and then assumes the binary state as applicable.  Unfortunate, that it is just about the only feature that you cannot determine the state of at need, i.e. you can for mute, volume level, application active, etc.  Kind of odd they don't address it, there is related discussion in WebOS development forum, if memory serves.
